File: Modulo 2/Clase03/app/src/main/java/com/area51/clase03/PersonaAdapter.java
package com.area51.clase03;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

public class PersonaAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    private ArrayList<Persona> lista;
    private Context context;

    public PersonaAdapter(ArrayList<Persona> lista, Context context) {
        this.lista = lista;
        this.context = context;
    }

    class PersonaViewHolder extends RecyclerView.ViewHolder {
        TextView tvNombre;
        TextView tvApellido;
        TextView tvGenero;

        public PersonaViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNombre = itemView.findViewById(R.id.tvNombre);
            tvApellido = itemView.findViewById(R.id.tvApellido);
            tvGenero = itemView.findViewById(R.id.tvGenero);
        }
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(context)
                .inflate(R.layout.item_persona, viewGroup, false);
        return new PersonaViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int i) {
        PersonaViewHolder personaViewHolder = (PersonaViewHolder) viewHolder;

        Persona persona = lista.get(i);
        personaViewHolder.tvNombre.setText(persona.getNombre());
        personaViewHolder.tvGenero.setText(persona.getGenero());
        personaViewHolder.tvApellido.setText(persona.getApellido());
    }

    @Override
    public int getItemCount() {
        return lista.size();
    }
}

File: Modulo 3/ClaseRecycler/app/src/main/java/com/area51/claserecycler/PersonaAdapter.kt
package com.area51.claserecycler

import android.app.AlertDialog
import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.area51.claserecycler.database.MetodoDBImpl
import kotlinx.android.synthetic.main.item_persona.view.*

class PersonaAdapter(val context: Context,
                     val lista: ArrayList<Persona>) :
        RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(context)
                .inflate(R.layout.item_persona, p0, false)
        return PersonaViewHolder(view)
    }

    class PersonaViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun init(persona: Persona, context: Context, lista: ArrayList<Persona>) {
            itemView.tvNombre.text = persona.nombre
            itemView.tvApellido.text = persona.apellido
            itemView.tvGenero.text = persona.genero

            itemView.contenedor.setOnClickListener {
                val dialog = AlertDialog.Builder(context)
                dialog.setTitle("Acción")
                dialog.setPositiveButton("Eliminar") { dialogInterface, i ->

                    val sql = MetodoDBImpl()
                    sql.eliminar(persona.id)
                    lista.remove(persona)

                }
                dialog.setNegativeButton("Actualizar") { dialogInterface, i ->

                }
                dialog.show()
            }
        }
    }

    override fun getItemCount(): Int {
        return lista.size
    }

    override fun onBindViewHolder(p0: RecyclerView.ViewHolder, p1: Int) {
        val holder = (p0 as PersonaViewHolder)
        holder.init(lista[p1], context, lista)
    }

}